backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/lsm/LiveMigrateDiskCommand.java

```diff
@@ -360,6 +362,11 @@ private void unlockDisk() {
         imageDao.updateStatusOfImagesByImageGroupId(getParameters().getImageGroupID(), ImageStatus.OK);
     }
 
+    private void unlockSnapshotLock() {
```

Comment:
To make things more "symmetric" and consistent, I would suggest to:
1. Rename it to `releaseSnapshotLock()`.
2. Add `acquireSnapshotLock()` and use it instead of lines 294-295.
```
private boolean acquireSnapshotLock() {
    EngineLock removeSnapshotLock = createEngineLockForSnapshotRemove();
    return lockManager.acquireLock(removeSnapshotLock).isAcquired());
}
```
3. Put both methods near `createEngineLockForSnapshotRemove()` below, in order to make a block of 3 related methods.

Reply:
ok, I wanted to rebase this PR so addressed the above
> 1. Rename it to `releaseSnapshotLock()`.

done

> 2. Add `acquireSnapshotLock()` and use it instead of lines 294-295.
> 
> ```
> private boolean acquireSnapshotLock() {
>     EngineLock removeSnapshotLock = createEngineLockForSnapshotRemove();
>     return lockManager.acquireLock(removeSnapshotLock).isAcquired());
> }
> ```

this part makes less sense - we may want to get the LockingResult as you pointed out before, in order to get the error message from it, right?

> 3. Put both methods near `createEngineLockForSnapshotRemove()` below, in order to make a block of 3 related methods.

done

Reply:
Agree, the suggested (2) above makes sense if we don't want to display the exact error message.
LGTM.